torch_xla/amp/autocast_mode.py

```diff
@@ -26,13 +27,15 @@ def __init__(self,
     self._xla_device = xm.xla_device_hw(device)
     if self._xla_device == 'GPU':
       backend = 'cuda'
+      self._cuda_bfloat16 = False
```

Comment:
I am very confuse what this `_cuda_bfloat16` actually means. Do you mind adding a comment above line 28 to explain all possible combinations and the expected behaviors?

Reply:
+1, realized that this variable should be called `_xla_bfloat16`. Let me add a brief comment, too.

Reply:
Something like this,
```
self._xla_bfloat16 = False  # True if xla backend with bfloat16 dtype.
if dtype is None:
   dtype = torch.float16
elif dtype == torch.bfloat16 and not torch.cuda.is_available():
   if xr.is_bf16_supported():
       # XLA:GPU with bfloat16 should run on `xla` backend
       # unless torch.autocast is compiled with cuda.
      backend = 'xla'
      self._xla_bfloat16 = True
```

Reply:
```
commit dc9224336af6cf6313e60d197dab47323e8e509d (HEAD -> spmd_amp_gpu, origin/spmd_amp_gpu)
Author: Yeounoh Chung <yeounoh@google.com>
Date:   Mon Sep 18 12:02:15 2023 -0700

    Rename _cuda_bfloat16 to _xla_bfloat16 since it is set when xla backend is used for bfloat16.

Reply:
Another confusing question is why there is only a special treatment for `torch.cuda.is_available()` is False and `dtype=bfloat16` here, is it because XLA:GPU itself doesn't support bfloat16? But when `torch.cuda.is_available() `is True, there is no special treatment for` bfloat16`, then the following is still used 
```
torch.set_autocast_xla_enabled(self.prev)
torch.set_autocast_xla_dtype(self.prev_dtype)
```

Reply:
Hi @baoleai `XLA:GPU` supports `bfloat16`, but we were using the `cuda` backend for autocast, when `torch.cuda.is_available()` was false. Instead we want to use `xla` backend.

When `torch.cuda.is_available()` is true, we use `cuda` backend for autocast, and since we are entering/existing the autocast context via `torch_xla.amp.autocast`, we still need to set 
```
torch.set_autocast_xla_enabled(self.prev)
torch.set_autocast_xla_dtype(self.prev_dtype)
```